spec/ddtrace/transport_spec.rb

```diff
@@ -0,0 +1,127 @@
+require 'spec_helper'
+
+require 'ddtrace'
+
+RSpec.describe Datadog::HTTPTransport do
+  let(:transport) { described_class.new('localhost', '8126', options) }
+  let(:options) { {} }
+
+  before(:each) do
+    @original_level = Datadog::Tracer.log.level
+    Datadog::Tracer.log.level = Logger::FATAL
+  end
+
+  after(:each) do
+    Datadog::Tracer.log.level = @original_level
+  end
+
+  describe '#handle_response' do
```

Comment:
probably worth checking here if the callback is processed

Reply:
Seems reasonable. Although, for what it's worth, the callback is removed from this function in the next PR (becomes a part of `post` instead in the next PR.)